tiledb/cloud/tiledb_cloud_error.py

```diff
@@ -40,8 +37,6 @@ def check_sql_exc(exc):
         raise Exception(internal_err_msg)
 
     try:
-        if client.TASK_ID_HEADER in exc.headers:
-            sql.last_sql_task_id = exc.headers[client.TASK_ID_HEADER]
```

Comment:
Can you explain why you don't want these here?

Reply:
Because this is already done in the `finally` block of `SQLResults.get`

Reply:
`check_sql_exc`/`check_udf_exc` are used in a few other places then the two try/except/final. It is probably worth leaving these for those other locations. What do you think?

Reply:
Makes sense, I'll add it back. In this case I'd prefer to remove the `finally`, set the `last_sql_task_id` only in the non-exception case and let this function set it on api exceptions. I find it more clear when things happen in exactly one place.

By the way note that the respective removed part in `check_udf_exc` was wrong; it was setting `cloudarray.last_sql_task_id` which doesn't exist (thus creating a new global in `cloudarray`).